src/pathDetails/PathDetails.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { useEffect, useRef, useState } from 'react'
+import styles from '@/pathDetails/PathDetails.module.css'
+import { HeightGraph } from 'leaflet.heightgraph/src/heightgraph'
+import 'leaflet.heightgraph/src/heightgraph.css'
+import { Path } from '@/api/graphhopper'
+import Dispatcher from '@/stores/Dispatcher'
+import { PathDetailsElevationSelected, PathDetailsHover, PathDetailsRangeSelected } from '@/actions/Actions'
+import { Coordinate } from '@/stores/QueryStore'
+
+interface PathDetailsProps {
+    selectedPath: Path
+}
+
+export default function ({ selectedPath }: PathDetailsProps) {
+    const containerRef: React.RefObject<HTMLDivElement> = useRef(null)
+    const [graph, setGraph] = useState<any | null>(null)
+    useEffect(() => {
+        const options = {
+            width: containerRef.current!.clientWidth,
+            height: containerRef.current!.clientHeight,
+            // todo: since we do not use this maybe we can/should remove the svg asset rules again because this we added
+            //       them just for this...
+            expandControls: false,
+        }
+        const callbacks = {
+            pointSelectedCallback: onPathDetailHover,
+            areaSelectedCallback: onRangeSelected,
+            routeSegmentsSelectedCallback: onElevationSelected,
+        }
+        setGraph(new HeightGraph(containerRef.current, options, callbacks))
+    }, [containerRef])
+    const resizeGraph = () => {
+        graph?.resize({ width: containerRef.current?.clientWidth, height: containerRef.current?.clientHeight })
+    }
+    useEffect(() => {
+        window.addEventListener('resize', resizeGraph)
+        return () => window.removeEventListener('resize', resizeGraph)
+    })
+    useEffect(() => {
+        const pathDetailsData = buildPathDetailsData(selectedPath)
+        graph?.setData(pathDetailsData.data, pathDetailsData.mappings)
+    }, [selectedPath, graph])
+    return (
```

Comment:
It would be nice to hide the height graph if no route is selected. It doesn't look pleasant when empty something like:

```
return isPathPresent && 
        <div className={styles.layoutContainer}>
            <div className={styles.heightgraphContainer} ref={containerRef} />
        </div>
```

Reply:
Sure. This is somewhat related to our discussion about using `null` for missing values. The `selectedPath` is never null, but it might be missing. I would find it very intuitive to check if its null here :) Currently it might be an `RouteStore#emptyPath`, but to obtain the `isPathPresent` value here I need to figure this out by checking if the points of the path are empty or something like this.

Reply:
I tried this but could not make it work yet :( I also wonder if we should or should not show the toggle button as in current master. I think it is useful to have the button so one can hide the graph if needed. Or maybe it should just be resizable in general? Previously I thought we would put the graph underneath the map so I removed the icon.

Reply:
I find a toggle button usefull.

Reply:
Ok I put back the toggle button, but I had to tweak the layout quite a bit for this. See my last commits

Reply:
The graph is no hidden when there is no path